### ISSUE TITLE
Errors/warnings in running demo

### ISSUE THREAD

**Issue by gevro** [gagneurlab/drop#40]
Hi, I'm getting various errors/warnings in running the demo. See below.

1)
/gpfs/share/apps/pandoc/2.2.3.2/bin/pandoc +RTS -K512m -RTS /tmp/Rtmp0annei/file5241f2818ed4f/Datasets.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash+smart --output /tmp/Rtmp0annei/file5241f11371564/Scripts_OUTRIDER_Datasets.html --email-obfuscation none --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --css lib/add_content_table.css --css lib/leo_style.css --variable 'theme:bootstrap' --include-in-header /tmp/Rtmp0annei/rmarkdown-str5241f5299b0a7.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/latex-div.lua --variable code_folding=hide --variable source_embed=Datasets.R --include-after-body /tmp/Rtmp0annei/file5241f340af0de.html --variable code_menu=1 
[WARNING] Could not parse YAML metadata at line 1 column 1: :8:122: Unexpected '

2)
[Mon May 18 11:47:58 2020]
rule markdown:
    input: aberrant_expression_readme.md
    output: /gpfs/scratch/evrong01/droptest/Output/htmlOutput/aberrant_expression_readme.html
    jobid: 3
[WARNING] This document format requires a nonempty <title> element.
  Please specify either 'title' or 'pagetitle' in the metadata.
  Falling back to 'aberrant_expression_readme'

3)
/gpfs/share/apps/pandoc/2.2.3.2/bin/pandoc +RTS -K512m -RTS /tmp/RtmpEilFDB/file52b51499c4c67/Datasets.utf8.md --to html4 --from markdown+autolink_bar
e_uris+tex_math_single_backslash+smart --output /tmp/RtmpEilFDB/file52b51182475e8/Scripts_Counting_Datasets.html --email-obfuscation none --standalone
 --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_s
mooth_scroll=1 --variable toc_print=1 --template /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/h/default.html --no-highlight --variable highl
ightjs=1 --css lib/add_content_table.css --css lib/leo_style.css --variable 'theme:bootstrap' --include-in-header /tmp/RtmpEilFDB/rmarkdown-str52b517a
25e6e6.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /gpfs/share/app
s/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/latex-div.lua --vari
able code_folding=hide --variable source_embed=Datasets.R --include-after-body /tmp/RtmpEilFDB/file52b517b12ba71.html --variable code_menu=1 
[WARNING] Could not parse YAML metadata at line 1 column 1: :9:121: Unexpected '

4)
[Mon May 18 11:48:38 2020]
rule create_graph:
    output: /gpfs/scratch/evrong01/droptest/Output/htmlOutput/AE_rulegraph.svg, /gpfs/scratch/evrong01/droptest/Output/htmlOutput/AE_rulegraph.png
    jobid: 1

WARNING: Less than 30 IDs in DROP_GROUP outrider
Structuring dependencies...
Dependencies file generated.

5)
Building DAG of jobs...
Error: Directory cannot be locked. This usually means that another Snakemake instance is running on this directory. Another possibility is that a prev
ious run exited unexpectedly.
WARNING: Less than 30 IDs in DROP_GROUP outrider
Structuring dependencies...
Dependencies file generated.

6)
Mon May 18 11:56:11 2020: Computing PCA ...
Mon May 18 11:56:11 2020: Fitting rho ...
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
6.000e-08 6.000e-08 6.000e-08 2.712e-02 3.189e-02 1.162e-01 
Mon May 18 11:56:13 2020: Writing final FRASER object ('/gpfs/scratch/evrong01/droptest/Output/processed_data/aberrant_splicing/datasets//savedObjects/fraser/fds-object.RDS').
Warning messages:
1: In injectOutliers(fds_copy, type = type, freq = injectFreq, minDpsi = minDeltaPsi,  :
  Injection-frequency is to low. Increasing it to `0.015` so we can inject at least 10 events into the data set!
2: In optimHyperParams(fds, type = type, implementation = implementation,  :
  No outliers could be injected so the hyperparameter optimization could not run. Possible reason: too few junctions in the data.
3: In injectOutliers(fds_copy, type = type, freq = injectFreq, minDpsi = minDeltaPsi,  :
  Injection-frequency is to low. Increasing it to `0.017` so we can inject at least 10 events into the data set!
4: In optimHyperParams(fds, type = type, implementation = implementation,  :
  No outliers could be injected so the hyperparameter optimization could not run. Possible reason: too few junctions in the data.
5: In injectOutliers(fds_copy, type = type, freq = injectFreq, minDpsi = minDeltaPsi,  :
  Injection-frequency is to low. Increasing it to `0.11` so we can inject at least 10 events into the data set!
Mon May 18 11:56:14 2020: Writing final FRASER object ('/gpfs/scratch/evrong01/droptest/Output/processed_data/aberrant_splicing/datasets//savedObjects/fraser/fds-object.RDS').

7)
/gpfs/share/apps/pandoc/2.2.3.2/bin/pandoc +RTS -K512m -RTS /tmp/RtmpT6Hg3C/file60a823d069667/Summary.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash+smart --output /tmp/RtmpT6Hg3C/file60a821c21ec95/fraser_countSummary.html --email-obfuscation none --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --css lib/add_content_table.css --css lib/leo_style.css --variable 'theme:bootstrap' --include-in-header /tmp/RtmpT6Hg3C/rmarkdown-str60a8261548510.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /gpfs/share/apps/R/3.6.3/lib64/R/library/rmarkdown/rmd/lua/latex-div.lua --variable code_folding=hide --variable code_menu=1 

Output created: /tmp/RtmpT6Hg3C/file60a821c21ec95/fraser_countSummary.html
Warning messages:
1: Transformation introduced infinite values in continuous y-axis 
2: Removed 380 rows containing missing values (geom_bar). 
3: Removed 75 rows containing non-finite values (stat_bin). 
4: Transformation introduced infinite values in continuous y-axis 
5: Removed 195 rows containing missing values (geom_bar). 
[1] TRUE TRUE TRUE TRUE

**Comment by mumichae** [comment]
Hi, could you show the general contents of your config file? It seems that some fields might be empty

**Comment by gevro** [comment]
Here it is. This is the standard drop demo. I didn't change anything.

projectTitle: Detection of RNA Outlier Pipeline
htmlOutputPath: /gpfs/scratch/evrong01/droptest/Output/htmlOutput
indexWithFolderName: true
fileRegex: .*\.R
root: /gpfs/scratch/evrong01/droptest/Output
sampleAnnotation: /gpfs/scratch/evrong01/droptest/Data/sample_annotation.tsv
geneAnnotation:
  v29: /gpfs/scratch/evrong01/droptest/Data/gencode_annotation_trunc.gtf
genomeAssembly: hg19
scanBamParam: null
aberrantExpression:
  groups:
  - outrider
  fpkmCutoff: 1
  implementation: autoencoder
  padjCutoff: 1
  zScoreCutoff: 0
  maxTestedDimensionProportion: 3
  dassie:
    tssWindow: 500
    pasWindow: 1000
aberrantSplicing:
  groups:
  - fraser
  recount: true
  longRead: false
  keepNonStandardChrs: true
  filter: false
  minExpressionInOneSample: 20
  minDeltaPsi: 0
  implementation: PCA
  padjCutoff: 1
  zScoreCutoff: 0
  deltaPsiCutoff: 0.05
  maxTestedDimensionProportion: 6
mae:
  groups:
  - mae
  genome: /gpfs/scratch/evrong01/droptest/Data/chr21.fa
  gatkIgnoreHeaderCheck: true
  padjCutoff: 0.05
  allelicRatioCutoff: 0.8
  addAF: false
  maxAF: 0.001
  qcVcf: /gpfs/scratch/evrong01/droptest/Data/qc_vcf_1000G.vcf.gz
  qcGroups:
  - mae
tools:
  gatkCmd: gatk
  bcftoolsCmd: bcftools
  samtoolsCmd: samtools

**Comment by vyepez88** [comment]
Hi,
all of those are warnings which don't prevent DROP from running.
1. The warning was due to an improper indentation in the header of one of the scripts. It's corrected now.
2. We specified the title in the readme file by doing:
`# Title`
we now changed it to:
```
---
title:
---
```
which gets rid of the warning.
3. Same as 1.
4. In general, we do not recommend running the pipeline with less than 30 samples per group. As the demo dataset has less than this number, it gives a warning on purpose. Nevertheless, it does run.
5. This is a `snakemake` attribute. When executing a process, `snakemake` locks the working directory. In this case, I guess you ran DROP with multiple cores which translated into parallel processes. `Snakemake` recognizes that there are many processes going on simultaneously in the same directory and throws that error. Nevertheless, the pipeline does not stop.
6. In order to find the best hyperparameters, FRASER injects outliers with a frequency of 0.1% and then tries to recall them. Due to the dataset being very small, the total number of injected outliers would be too small, therefore it has to use a higher frequency. The warning mentions this.
7. These warnings refer to the histograms showing the expression and variability of the filtered junctions. In order to better visualize those histograms, we plot them in the log10 scale on the y-axis. This makes the bins which have no values assigned to, to have an undefined, and therefore, missing value. Similar warnings happen in the other `Summary` scripts.
